src/net.h

```diff
@@ -712,6 +762,24 @@ class V1TransportSerializer  : public TransportSerializer {
     void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) override;
 };
 
+class V2TransportSerializer : public TransportSerializer
+{
+private:
+    std::unique_ptr<ChaCha20Poly1305AEAD> m_aead;
+    CPrivKey m_aead_k1; //keep the keys for a later rekeying
+    CPrivKey m_aead_k2;
+    uint64_t m_payload_seqnr = 0;
+    uint64_t m_aad_seqnr = 0;
+    int m_aad_pos = 0;
+
+public:
+    V2TransportSerializer(const CPrivKey& k1, const CPrivKey& k2) : m_aead(new ChaCha20Poly1305AEAD(k1.data(), k1.size(), k2.data(), k2.size())), m_aead_k1(k1), m_aead_k2(k2)
+    {
+    }
+    // prepare for next message
+    void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header);
```

Comment:
Add `override` keyword to be consistent with `V1TransportSerializer` (the same conditions apply here as well, no?)

Reply:
Right. Thanks. Fixed.